sdk/file_sdk/XMPFileFormat/xmp_file_format.cpp

```diff
@@ -38,10 +43,30 @@ void XMPFileFormat::Commit(shared_ptr<IStream> outputStream, string& newExtensio
   while (readCount > 0);
 
   outputStream->Seek(0);
-  XMPHelper::GetInstance().SetTags(outputStream, mTags);
+  XMPHelper::GetInstance().UpdateProperties(outputStream, mPropertiesToAdd, mkeysToRemove);
 
   newExtension = mExtension;
 }
 
+void XMPFileFormat::UpdateProperties(const std::vector<std::pair<std::string, std::string>>& propertiesToAdd, const std::vector<std::string>& keysToRemove)
+{
+  mPropertiesToAdd = propertiesToAdd;
+  mkeysToRemove = keysToRemove;
+
+  if(mHasProperties) {
+    for (vector<pair<string, string>>::iterator it=mProperties.begin(); it!=mProperties.end();)
+    {
+      string key = it->first;
+      if(std::any_of(mkeysToRemove.cbegin(), mkeysToRemove.cend(), [&key](const string& p) { return EqualsIgnoreCase(p, key);}))
+          it = mProperties.erase(it);
+      else
+          ++it;
+     }
+  }
+
+  mProperties.insert(mProperties.end(), mPropertiesToAdd.cbegin(), mPropertiesToAdd.cend());
+  mHasProperties = true;
```

Comment:
This will cause GetProperties not to do anything, meaning if no one called GetProperties - it will not return the properties from the file anymore.